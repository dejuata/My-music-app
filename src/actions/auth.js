import {
    getAuth,
    signInWithPopup,
    createUserWithEmailAndPassword,
    updateProfile,
    signInWithEmailAndPassword,
    signOut
} from 'firebase/auth';
import { googleAuthProvider } from '../firebase/firebaseConfig';
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';
import Swal from 'sweetalert2'


export const startLoginEmailPassword = (email, password) =>{

    return (dispatch) =>{

        dispatch( startLoading() );

        const auth = getAuth();
        signInWithEmailAndPassword(auth, email, password)
            .then(({ user }) => {
                console.log('dispatch login')
                dispatch(login(user.uid, user.displayName, 'firebase', ''))
            })
            .catch( (err) => {
                console.log(err.code, err.message);
                dispatch( finishLoading() );
                Swal.fire('Error', err.message, 'error')
            })

    }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {
    return (dispatch) => {
        const auth = getAuth();

        createUserWithEmailAndPassword(auth, email, password)
            .then( async ({ user }) => {

                await updateProfile(auth.currentUser, {
                    displayName: name,  photoURL: ""
                })

                dispatch(login(user.uid, user.displayName, 'firebase', ''))

            })
            .catch( (err) => {
                console.log(err.code, err.message);
                Swal.fire('Error', err.message, 'error')
            })

        console.log(auth.currentUser)
    }
}

export const startGoogleLogin = () =>{
    return (dispatch) =>{
        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider)
            .then( res => {
            })
    }
}

export const login = (uid, name, type, token) => ({
    type: types.login,
    payload: {
        uid,
        name,
        type,
        token
    }
})

export const startLogout = () => {
    return async (dispatch) => {
        const auth = getAuth();
        await signOut(auth)
        dispatch( logout() );
    }
}

export const logout = () => ({
    type: types.logout
})
