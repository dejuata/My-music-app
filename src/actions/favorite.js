import { types } from "../types/types";

export const setFavorites = (payload) => ({
    type: types.setFavorites,
    payload
})

export const removeFavorite= (payload) => ({
    type: types.removeFavorite,
    payload
})