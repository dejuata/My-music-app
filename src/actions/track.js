import { types } from "../types/types";

export const setTracks = (tracks) => ({
    type: types.setTracks,
    payload: tracks
})

export const addTrackFavorite = (id) => ({
    type: types.addTrackFavorite,
    payload: id
})

export const removeTrackFavorite = (id) => ({
    type: types.removeTrackFavorite,
    payload: id
})

export const logoutTrack = () => ({
    type: types.logoutTrack
})