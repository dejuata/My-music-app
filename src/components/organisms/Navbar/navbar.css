.o-navbar {
    display: flex;
    height: 50px;
    width: 100%;
    background-color: var(--color-white);
    justify-content: space-evenly;
}

.o-navbar div {
    height: 50px;
    width: 50px;
    /* border: 1px solid black; */
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    padding: 0.3rem;
    box-sizing: border-box;
    margin: 0 5px;
}

.o-navbar-logo {
    width: 90%;
    height: 90%;
}

.o-navbar__icon {
    color: var(--color-green);
}