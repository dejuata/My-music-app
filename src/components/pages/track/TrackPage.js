import React from 'react'
import { useParams } from 'react-router-dom';
import { useTrackDetail } from '../../../hooks/useTrackDetail';
import { TrackDetail } from '../../organisms/TrackDetail/TrackDetail';


export const TrackPage = () => {

    const { trackId } = useParams();

    const { data:track, loading } = useTrackDetail(trackId);


    return (
        <>
            {

                loading
                ?
                    <p>Loading</p>
                :
                    <TrackDetail {...track}/>

            }
        </>
    )
}
