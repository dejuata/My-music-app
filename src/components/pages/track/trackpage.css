.track-detail {
    width: 100%;
    /* background-color: var(--color-white); */
    /* color: var(--color-dark); */
    border-radius: 12px;
    padding: 12px;
    box-sizing: border-box;
}

.track-detail__image {
    margin: 0;
    /* padding: 20px; */
    /* box-sizing: border-box; */
}

.track-detail__image img{
    width: 100%;
    object-fit: cover;
    border-radius: 12px;
}

.track-detail__info {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-between;
    margin-top: 24px;
}

.track-detail__info h2 {
    margin: 0 ;
    font-size: var(--xlg);
    font-weight: bold;
}

.track-detail__info p {
    margin: 0;
}

.track-detail__info p:nth-child(2) {
    margin-bottom: 32px;
    font-size: var(--lg);

}

.track-detail__info p:nth-child(3) {
    font-size: var(--md);
    text-align: justify;
    font-weight: 300;
    margin-bottom: 12px;

}

.track-detail__info p:nth-child(4) {
    margin-bottom: 32px;
    font-weight: 300;
    font-size: var(--sm);
}

.secondary-button {
    background-color: var(--color-white);
    border-radius: 8px;
    border: 1px solid var(--color-text);
    color: var(--color-green);
    cursor: pointer;
    font-size: var(--lg);
    font-weight: 500;
    text-align: center;
    padding: 20px;
    box-sizing: border-box;
    width: 100%;
    text-decoration: none;
}