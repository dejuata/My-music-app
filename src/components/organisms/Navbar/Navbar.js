import React, { useContext } from 'react'
import { Link, NavLink, useNavigate } from 'react-router-dom'
import { FaHeart, FaPowerOff } from 'react-icons/fa';


import './navbar.css';
import { AuthContext } from '../../../auth/authContext';
import { authType } from '../../../types/authType';

export const Navbar = () => {

    const { user, dispatch } = useContext(AuthContext);
    const navigate = useNavigate();

    const logoPath = `/assets/logo.svg`;

    const handleLogout = () => {
        const action = {
            type: authType.logout
        }

        dispatch(action);

        navigate('/auth', {
            replace: true
        })
    }

    return (
        <nav className='o-navbar'>
            <NavLink to="/favorites">
                <div>
                    <FaHeart className='o-navbar__icon'/>
                </div>
            </NavLink>

            <NavLink to="/">
                <div>
                    <img src={logoPath} alt="logo" className='o-navbar-logo'/>
                </div>
            </NavLink>

            <div onClick={ handleLogout }>
                <FaPowerOff className='o-navbar__icon'/>
            </div>
        </nav>
    )
}
