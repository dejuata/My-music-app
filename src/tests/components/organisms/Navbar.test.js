import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { Navbar } from "../../../components/organisms/Navbar/Navbar";
import { startLogout } from '../../../actions/auth';
import { finishLoading } from '../../../actions/ui';
import { logoutTrack } from '../../../actions/track';

jest.mock('../../../actions/auth', () => ({
    startLogout: jest.fn()
}))

jest.mock('../../../actions/ui', () => ({
    finishLoading: jest.fn()
}))

jest.mock('../../../actions/track', () => ({
    logoutTrack: jest.fn()
}))


const middlewares = [ thunk ]
const mockStore = configureStore(middlewares)

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn()

const wrapper = mount(
    <Provider store={ store }>
        <MemoryRouter>
            <Navbar />
        </MemoryRouter>
    </Provider>
)

describe('test Navbar', () => {

    beforeEach( () => {
        store = mockStore(initState);
        // Limpiar mocks
        jest.clearAllMocks();
    })

    test('Should be render Navbar', () => {

        expect(wrapper).toMatchSnapshot();

    })


    test('Should be run the actions of handleLogout', () => {

        wrapper.find('FaPowerOff').simulate('click')

        expect( startLogout ).toHaveBeenCalledTimes(1);
        expect( finishLoading ).toHaveBeenCalledTimes(1);
        expect( logoutTrack ).toHaveBeenCalledTimes(1);

    })
});