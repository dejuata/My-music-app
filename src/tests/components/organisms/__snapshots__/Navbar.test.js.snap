// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navbar Should be render Navbar 1`] = `
<nav
  className="o-navbar"
>
  <a
    className=""
    href="/favorites"
    onClick={[Function]}
  >
    <div>
      <svg
        className="o-navbar__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
          key="0"
        />
      </svg>
    </div>
  </a>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <div>
      <svg
        className="o-navbar__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 576 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          key="0"
        />
      </svg>
    </div>
  </a>
  <div
    onClick={[Function]}
  >
    <svg
      className="o-navbar__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 54.1c63 45 104 118.6 104 201.9 0 136.8-110.8 247.7-247.5 248C120 504.3 8.2 393 8 256.4 7.9 173.1 48.9 99.3 111.8 54.2c11.7-8.3 28-4.8 35 7.7L162.6 90c5.9 10.5 3.1 23.8-6.6 31-41.5 30.8-68 79.6-68 134.9-.1 92.3 74.5 168.1 168 168.1 91.6 0 168.6-74.2 168-169.1-.3-51.8-24.7-101.8-68.1-134-9.7-7.2-12.4-20.5-6.5-30.9l15.8-28.1c7-12.4 23.2-16.1 34.8-7.8zM296 264V24c0-13.3-10.7-24-24-24h-32c-13.3 0-24 10.7-24 24v240c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24z"
        key="0"
      />
    </svg>
  </div>
</nav>
`;
